core/src/test/java/io/substrait/relation/AggregateRelTest.java

```diff
@@ -0,0 +1,162 @@
+
+  public static io.substrait.proto.NamedStruct createSchema() {
+
+    io.substrait.proto.Type i32Type =
+        io.substrait.proto.Type.newBuilder()
+            .setI32(io.substrait.proto.Type.I32.getDefaultInstance())
+            .build();
+
+    // Build a NamedStruct schema with two fields: col1, col2
+    io.substrait.proto.Type.Struct structType =
+        io.substrait.proto.Type.Struct.newBuilder().addTypes(i32Type).addTypes(i32Type).build();
+
+    return io.substrait.proto.NamedStruct.newBuilder()
+        .setStruct(structType)
+        .addNames("col1")
+        .addNames("col2")
+        .build();
+  }
+
+  public static io.substrait.proto.Expression createFieldReference(int col) {
+    // Build a ReferenceSegment that refers to struct field col
+    Expression.ReferenceSegment seg1 =
+        Expression.ReferenceSegment.newBuilder()
+            .setStructField(
+                Expression.ReferenceSegment.StructField.newBuilder().setField(col).build())
+            .build();
+
+    // Build a FieldReference that uses the directReference and a rootReference
+    Expression.FieldReference fieldRef1 =
+        Expression.FieldReference.newBuilder()
+            .setDirectReference(seg1)
+            .setRootReference(Expression.FieldReference.RootReference.getDefaultInstance())
+            .build();
+
+    // Wrap the FieldReference in an Expression.selection
+    return Expression.newBuilder().setSelection(fieldRef1).build();
+  }
+
+  @Test
+  public void testDeprecatedGroupingExpressionsAreMapped() {
+    Expression col1Ref = createFieldReference(0);
+    Expression col2Ref = createFieldReference(1);
+
+    AggregateRel.Grouping grouping =
+        AggregateRel.Grouping.newBuilder()
+            .addGroupingExpressions(col1Ref) // deprecated proto form
+            .addGroupingExpressions(col2Ref)
+            .build();
+
+    // Build an input ReadRel
+    ReadRel readProto = ReadRel.newBuilder().setBaseSchema(namedStruct).build();
+
+
+    // Build the AggregateRel with the new grouping_expressions field
+    AggregateRel aggrProto =
+        AggregateRel.newBuilder()
+            .setInput(Rel.newBuilder().setRead(readProto))
+            .addGroupings(grouping)
+            .build();
+
+    Rel relProto = Rel.newBuilder().setAggregate(aggrProto).build();
+    ProtoRelConverter converter = new ProtoRelConverter(functionLookup);
+    io.substrait.relation.Rel resultRel = converter.from(relProto);
+
+    assertTrue(resultRel instanceof Aggregate);
+    Aggregate agg = (Aggregate) resultRel;
+    assertEquals(1, agg.getGroupings().size());
+    assertEquals(2, agg.getGroupings().get(0).getExpressions().size());
+  }
+
+  @Test
+  public void testNewAggregateProtoForm() {
+    Expression col1Ref = createFieldReference(0);
+    Expression col2Ref = createFieldReference(1);
+
+    AggregateRel.Grouping grouping =
+        AggregateRel.Grouping.newBuilder()
+            .addExpressionReferences(0) // new proto form
```

Comment:
```suggestion
            .addExpressionReferences(0)
```
I don't think there is much value in calling this out here, and will create result in confusion when we eventually delete the old mechanism fully.